src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandler.java

```diff
@@ -86,92 +90,193 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
+                .forEach(sscsDocument -> sscsDocument.getValue().setDocumentFileName(sscsDocument.getValue().getDocumentLink().getDocumentFilename()));
+    }
+
     private void setMultiBundleConfig(SscsCaseData sscsCaseData, PreSubmitCallbackResponse<SscsCaseData> response) {
-        List<MultiBundleConfig> configs = new ArrayList<>();
-        
-        if (sscsCaseData.getDwpDocuments().stream().filter(f -> (f.getValue().getDocumentType().equals(DwpDocumentType.DWP_RESPONSE.getValue())
-                && f.getValue().getEditedDocumentLink() != null)
-                || f.getValue().getDocumentType().equals(DwpDocumentType.DWP_EVIDENCE_BUNDLE.getValue())
-                && f.getValue().getEditedDocumentLink() != null).count() > 0 &&  checkPhmeReviewIsGrantedOrUnderReview(sscsCaseData))  {
-
-            if (checkPhmeStatusIsUnderReview(sscsCaseData)) {
-                response.addError("There is a pending PHME request on this case");
-                return;
-            }
 
-            if (sscsCaseData.isLanguagePreferenceWelsh()) {
-                configs.add(MultiBundleConfig.builder().value(bundleWelshEditedConfig).build());
-                configs.add(MultiBundleConfig.builder().value(bundleWelshConfig).build());
-            } else {
-                configs.add(MultiBundleConfig.builder().value(bundleEnglishEditedConfig).build());
-                configs.add(MultiBundleConfig.builder().value(bundleEnglishConfig).build());
-            }
+        boolean hasEditedDwpResponseDocument = isHasEditedDwpResponseDocument(sscsCaseData);
 
-        } else {
-            if (sscsCaseData.isLanguagePreferenceWelsh()) {
-                configs.add(MultiBundleConfig.builder().value(bundleWelshConfig).build());
-            } else {
-                configs.add(MultiBundleConfig.builder().value(bundleEnglishConfig).build());
-            }
+        boolean hasEditedDwpEvidenceBundleDocument = isHasEditedDwpEvidenceBundleDocument(sscsCaseData);
+
+        if (!hasPhmeRequestOrConfidentialityUnderReview(sscsCaseData, response, hasEditedDwpResponseDocument, hasEditedDwpEvidenceBundleDocument)) {
+
+            sscsCaseData.setMultiBundleConfiguration(getMultiBundleConfigs(sscsCaseData));
         }
-        sscsCaseData.setMultiBundleConfiguration(configs);
     }
 
-    private boolean checkMandatoryFilesMissing(SscsCaseData sscsCaseData) {
-        if (null != sscsCaseData.getDwpDocuments()) {
+    private boolean hasPhmeRequestOrConfidentialityUnderReview(SscsCaseData sscsCaseData, PreSubmitCallbackResponse<SscsCaseData> response, boolean hasEditedDwpResponseDocument, boolean hasEditedDwpEvidenceBundleDocument) {
+        if (isPhmeStatusUnderReview(sscsCaseData) && (hasEditedDwpResponseDocument || hasEditedDwpEvidenceBundleDocument)) {
+            response.addError("There is a pending PHME request on this case");
+            log.info("There is a pending PHME request on this case");
+            return true;
```

Comment:
Would it be better to add both errors together (if both exist) so the user knows everything that is wrong with the case?

Reply:
changed